src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeInfo.cs

```diff
@@ -159,10 +163,13 @@ public static Task<SyntaxTreeDeclarationInfo> GetDeclarationInfoAsync(Document d
                             containsElementAccess = containsElementAccess || syntaxFacts.IsElementAccessExpression(node);
                             containsIndexerMemberCref = containsIndexerMemberCref || syntaxFacts.IsIndexerMemberCRef(node);
 
-                            DeclaredSymbolInfo declaredSymbolInfo;
-                            if (syntaxFacts.TryGetDeclaredSymbolInfo(node, out declaredSymbolInfo))
+                            if (!isGeneratedCode)
                             {
-                                declaredSymbolInfos.Add(declaredSymbolInfo);
+                                DeclaredSymbolInfo declaredSymbolInfo;
+                                if (syntaxFacts.TryGetDeclaredSymbolInfo(node, out declaredSymbolInfo))
+                                {
+                                    declaredSymbolInfos.Add(declaredSymbolInfo);
```

Comment:
The SyntaxTreeDeclarationInfo should probably always contain the real truth of what's defined in the syntax tree, regardless of whether the tree lives in a "generated" file, and then consumers of this can do their own filtering (e.g. in the NavigateToSymbolFinder).